From a7664b7bd52483826d0bd47b6a9b3394e6f6f1c6 Mon Sep 17 00:00:00 2001
From: Frajo Haider <f_haider@gmx.at>
Date: Mon, 7 Dec 2020 14:22:26 +0000
Subject: [PATCH 1/1] Force AVX2 off

---
 tcg/i386/tcg-target.inc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tcg/i386/tcg-target.inc.c b/tcg/i386/tcg-target.inc.c
index ae0228238b..7c4a83affc 100644
--- a/tcg/i386/tcg-target.inc.c
+++ b/tcg/i386/tcg-target.inc.c
@@ -3895,7 +3895,7 @@ static void tcg_target_init(TCGContext *s)
             asm(".byte 0x0f, 0x01, 0xd0" : "=a" (xcrl), "=d" (xcrh) : "c" (0));
             if ((xcrl & 6) == 6) {
                 have_avx1 = (c & bit_AVX) != 0;
-                have_avx2 = (b7 & bit_AVX2) != 0;
+                have_avx2 = false;//have_avx2 = (b7 & bit_AVX2) != 0;
             }
         }
     }
-- 
2.26.2

