From 456bf439bdaa2cefede8b09f224a660b8aee8353 Mon Sep 17 00:00:00 2001
From: Frajo Haider <f_haider@gmx.at>
Date: Tue, 8 Dec 2020 00:19:06 +0200
Subject: [PATCH 1/1] linux-user: disable commpage

---
 linux-user/elfload.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/linux-user/elfload.c b/linux-user/elfload.c
index fe9dfe795d..d18dca677a 100644
--- a/linux-user/elfload.c
+++ b/linux-user/elfload.c
@@ -4,6 +4,9 @@
 
 #include <sys/resource.h>
 #include <sys/shm.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <sys/utsname.h>
 
 #include "qemu.h"
 #include "disas/disas.h"
@@ -383,8 +386,10 @@ enum {
 
 /* The commpage only exists for 32 bit kernels */
 
-#define ARM_COMMPAGE (intptr_t)0xffff0f00u
+#define ARM_COMMPAGE 0
+#define init_guest_commpage() true
 
+#if 0
 static bool init_guest_commpage(void)
 {
     void *want = g2h(ARM_COMMPAGE & -qemu_host_page_size);
@@ -408,6 +413,7 @@ static bool init_guest_commpage(void)
     }
     return true;
 }
+#endif
 
 #define ELF_HWCAP get_elf_hwcap()
 #define ELF_HWCAP2 get_elf_hwcap2()
-- 
2.26.2

