From 3b522254716b6a1647716ebf4cb47d81da4734d4 Mon Sep 17 00:00:00 2001
From: Frajo Haider <f_haider@gmx.at>
Date: Wed, 29 Jan 2020 11:12:24 +0000
Subject: [PATCH 09/15] make sure mode is passed to openat if O_TMPFILE is
 specified. JB#45765

---
 linux-user/syscall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 91594472ed..5ec9a19ca2 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -396,7 +396,7 @@ static int sys_openat(int dirfd, const char *pathname, int flags, mode_t mode)
    * open(2) has extra parameter 'mode' when called with
    * flag O_CREAT.
    */
-  if ((flags & O_CREAT) != 0) {
+  if ((flags & O_CREAT) != 0 || ((flags) & O_TMPFILE) == O_TMPFILE) {
       return (openat(dirfd, pathname, flags, mode));
   }
   return (openat(dirfd, pathname, flags));
-- 
2.26.2

