From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frajo Haider <f_haider@gmx.at>
Date: Wed, 29 Jan 2020 11:12:24 +0000
Subject: [PATCH] make sure mode is passed to openat if O_TMPFILE is specified.
 JB#45765

---
 linux-user/syscall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 2e8504a4b3acac39b7a4777417c686f36b0ab1e3..9970e86702de58aa96c7b1e2bd5e65a2d0471d91 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -503,7 +503,7 @@ static int sys_openat(int dirfd, const char *pathname, int flags, mode_t mode)
    * open(2) has extra parameter 'mode' when called with
    * flag O_CREAT.
    */
-  if ((flags & O_CREAT) != 0) {
+  if ((flags & O_CREAT) != 0 || ((flags) & O_TMPFILE) == O_TMPFILE) {
       return (openat(dirfd, pathname, flags, mode));
   }
   return (openat(dirfd, pathname, flags));
