From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frajo Haider <f_haider@gmx.at>
Date: Tue, 8 Dec 2020 00:19:06 +0200
Subject: [PATCH] linux-user: disable commpage

---
 linux-user/elfload.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/linux-user/elfload.c b/linux-user/elfload.c
index 20894b633f58dcd8378a2a80e542a7313e517722..4b0e169e5ecf3053cd2aae11658978250df04613 100644
--- a/linux-user/elfload.c
+++ b/linux-user/elfload.c
@@ -417,8 +417,10 @@ enum {
 
 /* The commpage only exists for 32 bit kernels */
 
-#define HI_COMMPAGE (intptr_t)0xffff0f00u
+#define HI_COMMPAGE 0
+#define init_guest_commpage() true
 
+#if 0
 static bool init_guest_commpage(void)
 {
     abi_ptr commpage = HI_COMMPAGE & -qemu_host_page_size;
@@ -446,6 +448,7 @@ static bool init_guest_commpage(void)
                    PAGE_READ | PAGE_EXEC | PAGE_VALID);
     return true;
 }
+#endif
 
 #define ELF_HWCAP get_elf_hwcap()
 #define ELF_HWCAP2 get_elf_hwcap2()
